index.js

```diff
@@ -0,0 +1,219 @@
+            //sort languages
+            translatableEnglishLanguages.sort((a,b) => {
+                var nameA = a.name.toUpperCase(); // ignore upper and lowercase
+                var nameB = b.name.toUpperCase(); // ignore upper and lowercase
+                if (nameA < nameB) {
+                return -1;
+                }
+                if (nameA > nameB) {
+                return 1;
+                }
+            
+                // names must be equal
+                return 0;
+            })
+            const languagesMap = translatableEnglishLanguages.map( language => 
+                `<option value="${language.language}">${language.name}</option>`
+            )
+            $("#languageSelector").html(languagesMap)
+        })
+    }
+
+
+  //this method translates the text using the `translate` enpoint created. 
+    function translateText(message, language, messageType = "receive") {
+        //check if the message is a sent message or received message
+        const text = messageType === "send" ? message: message.data;
+        const translateParams = {
+            text: text,
+            modelId: messageType === "send" ? `${language}-en` : `en-${language}`,
+        };
+        var nmtValue  = '2019-09-28';
+        fetch('/api/translate', {
+            method: 'POST',
+            body: JSON.stringify(translateParams),
+            headers: new Headers({
+                'X-WDC-PL-OPT-OUT': $('input:radio[name=serRadio]:radio:checked').val(),
+                'X-Watson-Technology-Preview': nmtValue,
+                "Content-Type": "application/json"
+            }),
+        })
+        .then(response => response.json())
+        .then(data => {
+            
+            const translatedText = data['translations'][0]['translation'];
+            if ( messageType === "send") {
+                channel.publish('text', translatedText);
+                channel.publish("user", {
+                    "name": user.name,
+                    "avatar": user.avatar
+                });
+            } else {
+                show(translatedText, message.timestamp, otherUser);
+            }
+        })
+        .catch(error => console.error(error)) 
+    }
+
+
+    const ably = new Ably.Realtime({
+        key: YOUR_ABLY_API_KEY,
+        clientId:`${user.id}`,
+        echoMessages: false
+    });
+
+    //specify the channel the user should belong to. In this case, it is the `test` channel
+    const channel = ably.channels.get('test');
+
+    //Subscribe the user to the messages of the channel. So the use rwill receive each message sent to the test channel.
+
+    channel.subscribe("text", function(message) {
+        const selectedLanguage = $("#languageSelector").find(":selected").val();
+        translateText(message, selectedLanguage)
+    });
+
+    //This gets the data of other users as they publish to the channel.
+    channel.subscribe("user", (data) => {
+        var dataObj = JSON.parse(JSON.stringify(data));
```

Comment:
The data object is already in JSON format by default. Can you update this in a new commit please? No need to do a new step-wise commit.

Reply:
Please also update the tutorial one as well after having tested it, thanks.